bloom-housing/bloom
# fix: add email-service

# Pull Request Template

## Issue Overview

This PR addresses #3503


## Description

This add all of the endpoints in the EmailService and connects them up to appropriate use cases throughout the app.

Changes from the current backend's Email service:
- Jurisdiction is no longer acquired from the request header and rather from the context within the function calling the email service such as the user account.
- Figuring out which translation to get has been moved to the translation service
- Email Service is no longer injected at Request Scope. This could potentially cause issues if two people are requesting emails at the same time. It will require additional testing in a better environment to know if that is the case.

Note:
- Send MFA code and application confirmation emails functions are setup, but not connected to any controller since they are dependent on other services that are not completed

## How Can This Be Tested/Reviewed?

Make sure the following env variables are added to the .env file:
```
PARTNERS_PORTAL_URL
EMAIL_API_KEY
```
Also make sure you do a reseed to get the translations in the database. 

The following should be tested:
- Create Public User using the `/user` post endpoint. You should receive the "Welcome" email
- Create Partner user by using the `/user/invite` endpoint. You should receive the "Invite" email
- Resend public confirmation email using the `/user/resend-confirmation` endpoint.
  - You can use the user created in the first use case as long as you haven't confirmed the user
- Resend the partner user confirmation email using the `/user/resend-partner-confirmation` endpoint
   - You can use the user created in the second use case as long as you haven't confirmed the user
- Get a forgot password email when hitting the `/user/forgot-password` endpoint
- Change a user email by updating the user via the `/user/{id}` put endpoint. Make sure you put a new email in the request

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
